crates/agglayer-rpc/src/lib.rs

```diff
@@ -194,27 +271,206 @@ where
+        let settled_height = latest_settled_certificate.as_ref().map(|cert| cert.height);
+        let settled_certificate_id = latest_settled_certificate
+            .as_ref()
+            .map(|cert| cert.certificate_id);
+
+        // Extract settled_pp_root from the settled certificate's proof public values
+        let settled_pp_root = latest_settled_certificate.as_ref().and_then(|cert| {
+            // Get the proof for the settled certificate
+            self.get_proof(cert.certificate_id)
+                .inspect_err(|error| {
+                    error!(
+                        ?error,
+                        "get network status: failed to get proof for settled certificate"
+                    );
+                })
+                .ok()
+                .flatten()
+                .and_then(|proof| {
+                    // Deserialize the proof's public values to get PessimisticProofOutput
+                    let agglayer_types::Proof::SP1(sp1_proof) = proof;
+                    pessimistic_proof::PessimisticProofOutput::bincode_codec()
+                        .deserialize::<pessimistic_proof::PessimisticProofOutput>(
+                            sp1_proof.public_values.as_slice(),
+                        )
+                        .inspect_err(|error| {
+                            error!(
+                                ?error,
+                                "get network status: failed to deserialize pessimistic proof \
+                                 output"
+                            );
+                        })
+                        .ok()
+                        .map(|output| output.new_pessimistic_root)
+                })
+        });
+
+        let settled_ler = latest_settled_certificate
+            .as_ref()
+            .map(|cert| cert.new_local_exit_root);
+
+        let settled_let_leaf_count = self
+            .state
+            .read_local_network_state(network_id)
+            .inspect_err(|error| {
+                error!(
+                    ?error,
+                    "Failed to get local network state for network {network_id}"
+                );
+            })
+            .ok()
+            .flatten()
+            .map(|local_network_state| {
+                // We return the leaf count of the latest local exit tree
+                local_network_state.exit_tree.leaf_count as u64
+            });
+
+        let latest_pending_height = latest_pending_certificate
+            .as_ref()
+            .map(|cert| cert.height.as_u64());
+
+        let latest_pending_status = latest_pending_certificate
+            .as_ref()
+            .map(|cert| cert.status.clone());
+
+        // Get pending certificate error if exists
+        let latest_pending_error = latest_pending_certificate
+            .as_ref()
+            .and_then(|cert| match &cert.status {
+                agglayer_types::CertificateStatus::InError { error } => Some(*error.clone()),
```

Comment:
```suggestion
                agglayer_types::CertificateStatus::InError { error } => Some((*error).clone()),
```
Clone the value inside the box directly, instead of cloning the whole box and then copying out. Might be optimized away by the compiler but :shrug: